innopoints/views/product.py

```diff
@@ -101,8 +105,18 @@ def create_product():
         log.exception(err)
         abort(400, {'message': 'Data integrity violated.'})
 
-    out_schema = ProductSchema(exclude=('notifications',
-                                        'varieties.product_id',
+    # TODO: replace the following with proper debounce
+    # Check if a notification has been sent today
+    query = Notification.query.filter(
+        Notification.type == 'new_arrivals',
+        Notification.timestamp >= date.today()
+    )
+    if query.count() == 0:
```

Comment:
There's an `exists` type of query. It should be more efficient. You may find a sample [here](https://github.com/illright/innopoints-backend/blob/master/innopoints/models/activity.py#L83-L84). But if we are to rewrite this anyway, maybe it doesn't matter after all.

Reply:
I saw it, but I thought `db.session.query(query.exists()).scalar()` was a mouthful. Is it somehow more efficient to be worth changing?

Reply:
I dare say it is: https://blog.jooq.org/2016/09/14/avoid-using-count-in-sql-when-you-could-use-exists/

I mean, yeah, premature optimization and shit, this is not worth it unless we have real data to back this optimization, but this one just seems like a no brainer to me